### ISSUE TITLE
CORS is the devil 😈

### ISSUE THREAD

**Issue by MattPereira** [MattPereira/ccgc-backend#2]
### Overview
The first few requests to the back-end of my app are `blocked by CORS policy`. The initial request also always crashes the deployment on render. After render has time to restart the crashed server, the site functions properly with no CORS issues until it is given inactive time to fall back asleep. Is it render cloud or is it my code?

Front-end deployed on vercel: https://ccgc.vercel.app/

Back-end deployed on render: https://ccgc-backend.onrender.com/

Relevant code for CORS: https://github.com/MattPereira/ccgc-backend/blob/main/app.js

### Screenshots

<details><summary>Errors in browser console</summary>

![image](https://user-images.githubusercontent.com/73561520/229315450-fb9d3234-b4c3-496c-804e-3cb3d302055b.png)

</details>

<details><summary>Server logs on render ( as it crashes on first request and then restarts )</summary>

![image](https://user-images.githubusercontent.com/73561520/229315431-537044a5-80e9-4d4e-a15f-7482c5ea6e42.png)

</details>

<details><summary>Network tab</summary>

![image](https://user-images.githubusercontent.com/73561520/229315412-c833ad22-0aad-45c3-a599-2073fa806420.png)

</details>

<details><summary>Hitting sleepy API from insomnia/postman</summary>

![image](https://user-images.githubusercontent.com/73561520/229325656-1280db81-b796-46bd-9e87-07e46db07e10.png)

</details>

### Details
- The server crashes on the first request if its been a while, but is okay after it has time to restart until it goes back to sleep again
- I am using this npm package for cors : https://www.npmjs.com/package/cors
- I am paying for "Starter" $7 a month web service from [Render](https://render.com/docs/web-services)
- The PostgreSQL database is on https://railway.app/

### Resources
- https://render.com/docs
- https://www.npmjs.com/package/cors
- https://expressjs.com/en/resources/middleware/cors.html

**Comment by MattPereira** [comment]
github thread on unhandled pg connection error
https://github.com/brianc/node-postgres/issues/1611

**Comment by MattPereira** [comment]
Fixed the cors part. starting new issue for pg connection after idle
